athena-vertica/src/main/java/com/amazonaws/connectors/athena/vertica/VerticaJdbcSplitQueryBuilder.java

```diff
@@ -0,0 +1,297 @@
+ * Query builder for database table split.
+ */
+public class VerticaJdbcSplitQueryBuilder {
+    private static final Logger LOGGER = LoggerFactory.getLogger(VerticaJdbcSplitQueryBuilder.class);
+    private static final String EXPORT_BUCKET = "export_bucket";
+    private final String quoteCharacters;
+
+    /**
+     * @param quoteCharacters database quote character for enclosing identifiers.
+     */
+      public VerticaJdbcSplitQueryBuilder(String quoteCharacters) {
+        this.quoteCharacters = Validate.notBlank(quoteCharacters, "quoteCharacters must not be blank");
+    }
+
+    /**
+     * Common logic to build Split SQL including constraints translated in where clause.
+     *
+     * @param jdbcConnection JDBC connection. See {@link Connection}.
+     * @param catalog        Athena provided catalog name.
+     * @param schema         table schema name.
+     * @param table          table name.
+     * @param tableSchema    table schema (column and type information).
+     * @param constraints    constraints passed by Athena to push down.
+     * @param split          table split.
+     * @return prepated statement with SQL. See {@link PreparedStatement}.
+     * @throws SQLException JDBC database exception.
+     */
+    public String buildSql(
+            final String schema,
+            final String table,
+            final Schema tableSchema,
+            final Constraints constraints,
+            final String queryID)
+            throws SQLException {
+
+        //get the bucket where export results wll be uploaded
+        String s3ExportBucket = System.getenv(EXPORT_BUCKET);
+        StringBuilder sql = new StringBuilder();
+
+        String exportSqL = "EXPORT TO PARQUET(" +
+                "directory = 's3://"+ s3ExportBucket + "/"+queryID.replace("-","")+ "'" +
+                ", Compression='snappy'\n" +
+                ", fileSizeMB=1000) \n" +
+                "AS \n";
+
+        sql.append(exportSqL);
+
+        //get the column names to be queried
+        String columnNames = tableSchema.getFields().stream()
+                .map(Field::getName)
+                .map(this::quote)
+                .collect(Collectors.joining(", "));
+
+        sql.append("SELECT ");
+        sql.append(columnNames);
+
+        if (columnNames.isEmpty()) {
+            sql.append("null");
+        }
+
+        sql.append(getFromClauseWithSplit(schema, table));
+
+        HashMap<String,TypeAndValue> accumulator = new HashMap<>();
+
+        List<String> clauses = toConjuncts(tableSchema.getFields(), constraints, accumulator);
+
+        if (!clauses.isEmpty()) {
+            sql.append(" WHERE ")
+                    .append(Joiner.on(" AND ").join(clauses));
+        }
+
+        //wrap the export around the generated sql
+        LOGGER.info("Generated SQL without the constraints : {}", sql.toString());
+
+        //Using StringTemplates to fill in the values of constraints
+        ST sqlTemplate = new ST(sql.toString());
```

Comment:
now i'm really confused :) using string template but still doing a good amount of String concatenation above. Why? Let me know if you need some examples of how to use StringTemplate to build SQL. We have a few in-progress connectors that use it and I can share links with you internally.